lib/modules/datasource/rubygems/versions-datasource.ts

```diff
@@ -96,18 +106,30 @@ export class VersionsDatasource extends Datasource {
   }
 
   /**
-   * https://bugs.chromium.org/p/v8/issues/detail?id=2869
+   * Since each `/versions` reponse exceed 10MB,
+   * there is potential for a memory leak if we construct slices
+   * of the response body and cache them long-term:
+   *
+   *   https://bugs.chromium.org/p/v8/issues/detail?id=2869
+   *
+   * This method meant to be called for `version` and `packageName`
+   * before storing them in the cache.
    */
-  private static copystr(x: string): string {
-    return (' ' + x).slice(1);
+  private copystr(x: string): string {
+    const len = Buffer.byteLength(x, 'utf8');
+    const buf = this.isInitialFetch
+      ? Buffer.allocUnsafe(len) // allocate from pre-allocated buffer
+      : Buffer.allocUnsafeSlow(len); // allocate standalone buffer
```

Comment:
what's the difference and why do we need this condition?

Reply:
During the first run, we'll call the `copystr` a lot, so the most of memory chunks will be co-located in the pre-allocated 4Kb buffers. During the further updates, we want to avoid such co-location, since it may share single buffer with some other unrelated data, that can complicate GC.

Reply:
At least, this is how I understand it

Reply:
ok